### PR TITLE
Mothroach Cargo Crates

### PR DESCRIPTION
**Pull request by BYONDFuckery**
Requires #21855
## About the PR
Added Mothroach Crates, orderable in Cargo.

## Why / Balance
I wanted Mothroach Crates in the game because I could never find them in maints. WYCI and all.

## Technical details
Changed some YML around.

## Media

-  [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase [Video Here](https://drive.google.com/file/d/1OnoHbI8Hytip2VavtW5kr9qDd-C5G24-/view?usp=sharing)

## Breaking changes

**Changelog**

:cl: Avalon
- add: CentCom has begun unceremoniously plucking Mothroaches out of Maintenance and selling them to nearby Space Stations after the discovery that they make decent pets.

### CODE REVIEW

**Comment by BYONDFuckery** [comment]
Reason I deleted the .yml was because there were some extras in master branch so I removed them because nothing needed them

**Comment by BYONDFuckery** [comment]
It wasn't the ACTUAL .yml, just copies

**Comment by CatMagic1** [comment]
goobers rejoice

**Comment by VasilisThePikachu** [comment]
6000 spasos for 4 mothroaches seems like a bit... much

**Comment by BYONDFuckery** [comment]
> 6000 spasos for 4 mothroaches seems like a bit... much

What do you think? 4k?

**Comment by asperger-sind** [comment]
> > 6000 spasos for 4 mothroaches seems like a bit... much
> 
> What do you think? 4k?

500 and then bump it to 3k after someone clips a tide of mothroaches and posts it to #screenshots-videos

**Comment by VasilisThePikachu** [comment]
like 1-2k maybe??

**Comment by BYONDFuckery** [comment]
> like 1-2k maybe??

Let me commit that

**Comment by BYONDFuckery** [comment]
I'm gonna revert that.

**Comment by BYONDFuckery** [comment]
> I'm gonna revert that.

For some weird ass reason this makes something go wrong.

File: Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml [hunk 1]
```diff
@@ -197,3 +197,13 @@
   cost: 2800
   category: Livestock
   group: market
+
+- type: cargoProduct
+  id: LivestockMothroach
+  icon: mothroach.rsi
```
Comment (TheShuEd) [inline Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:203]: remove mothroach.rsi after `icon:`

File: Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml [hunk 2]
```diff
@@ -197,3 +197,13 @@
...
+    sprite: Mobs/Animals/mothroach.rsi
+    state: mothroach
+  product: CrateNPCMothroach
+  cost: 2500
```
Comment (TheShuEd) [inline Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:207]: reduce the value of the moths. They're cute, but not so cute that they cost so much money.)

**Review by TheShuEd — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml (hunk 2 above)
Comment (CatMagic1) [reply Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:207]: the mothconomy will never recover

Comment (MouseHatsForAll) [reply Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:207]: 3500 seems more fair, tbh

File: Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml (hunk 1 above)
Comment (MouseHatsForAll) [reply Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:203]: yeah, he's right. You gotta just leave it as `icon:`
aside from that, it should be good.

**Comment by BYONDFuckery** [comment]
i do not understand merge issues :godo:

**Comment by BYONDFuckery** [comment]
checks work which is a plus but

**Comment by BYONDFuckery** [comment]
image

**Comment by BYONDFuckery** [comment]
> image

conflicts are coming from here

File: Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl [hunk 3]
```diff
@@ -53,4 +53,7 @@ ent-CrateNPCLizard = Lizard crate
     .desc = A crate containing a lizard.
 
 ent-CrateNPCKangaroo = Kangaroo crate
-    .desc = A crate containing a kangaroo.
\ No newline at end of file
+    .desc = A crate containing a kangaroo.
+
+ent-CrateNPCMothroach = Crate of mothroaches
+    .desc = A crate containing four mothroaches. You can hear faint chittering inside.
```
Comment (MouseHatsForAll) [inline Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl:59]: From what I'm understanding, it seems like the extra sentence is throwing it off, somehow. Just change line 59 to be `.desc = A crate containing four mothroaches.` and make line 60 empty, and I think that might fix it.

Comment (MouseHatsForAll) [inline Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl:59]:
```suggestion
    .desc = A crate containing four mothroaches.
    
```

**Comment by lzk228** [comment]
check up on the internet what is merge conflict on guthub

**Review by MouseHatsForAll — APPROVED** [review]
I think that should fix it. Run it through the tests, but as far as I can tell it should be good.

File: Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl (hunk 3 above)
Comment (BYONDFuckery) [reply Resources/Locale/en-US/prototypes/catalog/fills/crates/livestock-crates.ftl:59]: I like the chittering part.

**Comment by BYONDFuckery** [comment]
Merge conflicts fixed. waiting on checks. looks good

**Comment by BYONDFuckery** [comment]
image
why

**Comment by VasilisThePikachu** [comment]
> image why

In the logs of the test run

``Found arbitrage on LivestockMothroach cargo product! Cost is 2500 but sell is 5285!``

**Comment by BYONDFuckery** [comment]
ah so sell 2500
infinite money meta avoided.

**Comment by BYONDFuckery** [comment]
> > image why
> 
> In the logs of the test run
> 
> `Found arbitrage on LivestockMothroach cargo product! Cost is 2500 but sell is 5285!`

Does anyone know how exactly I set the sell price?

**Comment by lzk228** [comment]
> 
> Does anyone know how exactly I set the sell price?

it is calculated from crate's content

**Comment by BYONDFuckery** [comment]
> > Does anyone know how exactly I set the sell price?
> 
> it is calculated from crate's content

is there a way to force it or do i just have to have the moths at 5k

**Comment by lzk228** [comment]
mothroach price is 60
https://github.com/space-wizards/space-station-14/blob/641b4903135a107f0cb1c1cb78bab8c1843dab67/Resources/Prototypes/Entities/Mobs/NPCs/animals.yml#L426
but they so expensive because they have moth body, so make #21753 fixed, or increase crate price

File: Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml (hunk 2 above)
Comment (BYONDFuckery) [reply Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:207]: i have to do it anyway but youll get 6 moths per crate now

**Comment by BYONDFuckery** [comment]
> mothroach price is 60
> 
> https://github.com/space-wizards/space-station-14/blob/641b4903135a107f0cb1c1cb78bab8c1843dab67/Resources/Prototypes/Entities/Mobs/NPCs/animals.yml#L426
> 
> but they so expensive because they have moth body, so make #21753 fixed, or increase crate price

it's still broken :mistaken:

**Comment by VasilisThePikachu** [comment]
theres really no point to increasing the price imo, just fix #21753

**Comment by BYONDFuckery** [comment]
i will be sad if this fails :+1:

**Comment by BYONDFuckery** [comment]
**uh oh**

**Comment by BYONDFuckery** [comment]
***okay oh no***

**Comment by BYONDFuckery** [comment]
***i bet it's broken*** :-1:

**Comment by BYONDFuckery** [comment]
***WAIT THIS MIGHT WORK***

**Comment by BYONDFuckery** [comment]
image

**Comment by BYONDFuckery** [comment]
https://github.com/space-wizards/space-station-14/issues/21753 moment

**Comment by VasilisThePikachu** [comment]
Fun fact: you can run tests locally with this command

``dotnet test -e COMPlus_gcServer=1``

Then you can test changes yourself!

As for your issue, can you not just remove ``prototype: Moth`` from the mothroaches?

**Comment by lzk228** [comment]
> As for your issue, can you not just remove `prototype: Moth` from the mothroaches? Idk what kind of bugs it will cause if any but try it

poor mothroaches won't be able to eat clothes

**Comment by VasilisThePikachu** [comment]
> > As for your issue, can you not just remove `prototype: Moth` from the mothroaches? Idk what kind of bugs it will cause if any but try it
> 
> poor mothroaches won't be able to eat clothes

Then give them the right components manually?

**Comment by VasilisThePikachu** [comment]
Requires #21855 to be merged first.

**Comment by BYONDFuckery** [comment]
thanks to everyone who wrote 90% of my code :)

**Comment by BYONDFuckery** [comment]
image
my anxiety thinks im gonna fat finger and press this

**Comment by lzk228** [comment]
bro stop doing useless comments lol

**Comment by BYONDFuckery** [comment]
> bro stop doing useless comments lol

absolutely not. this is my pull request. yield to my codery
(this is a joke of course)

**Comment by BYONDFuckery** [comment]
still errors

**Comment by BYONDFuckery** [comment]
god fucking damn it
https://github.com/space-wizards/space-station-14/pull/21855

**Comment by BYONDFuckery** [comment]
Drafted due to requirements.

**Comment by BYONDFuckery** [comment]
THIS IS SO FUCKING GREAT
HOLY SHIT

File: Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml (hunk 2 above)
Comment (BYONDFuckery) [reply Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:207]: reverted

**Comment by BYONDFuckery** [comment]
me getting github to re-check it

**Review by DrSmugleaf — APPROVED** [review]

**Comment by sheepism** [comment]
ROUND 2 FIGHT!!!

**Pull request merged**